### PR TITLE
Start the eventNonce counter at last observed

### PR DESCRIPTION
**Pull request by jkilpatr**
Previously the last seen event nonce for an ethereum oracle always
started at zero.

if a new validator joined the set and had never submitted
an event before they would need to play back the entire event history
in order to catch up.

From an efficiency standpoint this could involve searching over
millions of blocks and submitting thousands or tens of thousands
of oracle messages.

This patch resolves that issue starting any validator who has never
submitted an oracle event at the last observed nonce. This isn't
correct as it actually needs to be the last observed event that is
not yet past the slashing window.

### CODE REVIEW

**Comment by jtremback** [comment]
The problem that this PR is trying to solve arises only in `storeClaim`:

```golang
// storeClaim persists a claim. Fails when a claim submitted by an Eth signer does not increment the event nonce by exactly 1.
func (k Keeper) storeClaim(ctx sdk.Context, details types.EthereumClaim) error {
	// Check that the nonce of this event is exactly one higher than the last nonce stored by this validator.
	// We check the event nonce in processAttestation as well, but checking it here gives individual eth signers a chance to retry.
	lastEventNonce := k.GetLastEventNonceByValidator(ctx, sdk.ValAddress(details.GetClaimer()))
```
However, it may not be necessary to even check the nonce in this function. As outlined in https://github.com/althea-net/peggy/issues/119, the nonce check in `storeClaim` does not guarantee that attestations will be Observed and accepted into the Cosmos state in order:

> When an individual validator submits a claim about an event, we error if their previously submitted claim does not have a nonce exactly 1 lower than the current claim. This allows honest validators to correct mis-orderings caused by network delay and other issues.
>
> However, a validator can submit any event under a certain nonce. A malicious validator could vote for a different event with nonce 2, then vote for the same event with nonce 3 as everyone else. This could cause event 3 to pass the threshold before event 2. This could probably also happen naturally, without malicious validators, although it seems unlikely.

For this reason, we also check the contiguousness of nonces of Observed attestations in `processAttestation`. The code in `processAttestation` is the real safeguard (and needs to be improved as noted in #119). The nonce-checking code in `storeClaim` really only serves to notify a validator of problems on their end. It adds no security.

In conclusion, the nonce-checking code in `storeClaim` which has the problem that this PR is trying to solve does not add anything to Peggy's security. Perhaps it should be removed instead of fixing it. Also, we've got to get back on fixing #119

**Comment by jkilpatr** [comment]
@jtremback  and I discussed this and agreed that the contiguous nonce requirement stands. Mostly as a way to prevent double voting, where a validator could vote for multiple attestations on the same nonce. Obviously we could add protections for double voting but doing so without enforcing ordering would be at least as complex as this change anyway.
